### PR TITLE
OKTA-421213 dev docs for dynamic issuer mode

### PR DESCRIPTION
**Pull request by mymoonasaifunnissameerasahib-okta**
OKTA-421213 dev docs for dynamic issuer mode

## Description:
- **dev docs for dynamic issuer mode** 
- **Is this PR related to a Monolith release? Yes 2021.09.0** 

### Resolves:

* [OKTA-421213](https://oktainc.atlassian.net/browse/OKTA-421213)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 1]
```diff
@@ -1328,7 +1328,7 @@ Adds an OAuth 2.0 client application. This application is only available to the
...
+* `issuer_mode` <ApiLifecycle access="ea" /> is visible if the Custom URL Domain feature is enabled. If the feature is enabled, you can set a custom domain URL in the settings for an OpenID Connect token in an app, and this property is returned in the appropriate responses. After the feature is enabled, the default value for new apps is `CUSTOM_URL`. For existing apps, the default remains `ORG_URL`. You can also update this to `DYNAMIC` if the Custom URL Domain feature is enabled. If it is set to `DYNAMIC` the request for the ID token made from Custom Domain will have the Custom URL as the issuer and the request for the ID token made from Okta Org Domain will have the Org URL as the issuer. You can change the value using the API or administrator user interface. To enable the Custom URL Domain feature, contact [Support](https://support.okta.com/help/open_case).
```
Comment (yuliu-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:1331]: Isn't the default issuer mode for new client will always be "DYNAMIC"?

File: packages/@okta/vuepress-site/docs/reference/api/authorization-servers/index.md [hunk 2]
```diff
@@ -161,6 +161,8 @@ When you use these API endpoints to create or modify an Authorization Server res
 
 * If set to `CUSTOM_URL`, then in responses, `issuer` is the custom domain URL configured in the administration user interface.
 
+* If set to `DYNAMIC`, then in responses, `issuer` is the custom domain URL if the token was issued by custom domain or the Okta org's original domain URL if the token was issued by the original Okta org domain.
```
Comment (yuliu-okta) [inline packages/@okta/vuepress-site/docs/reference/api/authorization-servers/index.md:164]: `issuer` is the custom domain URL if the oauth2 request was sent to the custom domain or the Okta domain URL if the oauth2 request was sent to the original Okta org domain.

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md (hunk 1 above)
Comment (mymoonasaifunnissameerasahib-okta) [reply packages/@okta/vuepress-site/docs/reference/api/apps/index.md:1331]: Only if the dynamic issuer mode is enabled.

File: packages/@okta/vuepress-site/docs/reference/api/authorization-servers/index.md (hunk 2 above)
Comment (mymoonasaifunnissameerasahib-okta) [reply packages/@okta/vuepress-site/docs/reference/api/authorization-servers/index.md]: updated

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md (hunk 1 above)
Comment (mymoonasaifunnissameerasahib-okta) [reply packages/@okta/vuepress-site/docs/reference/api/apps/index.md]: updated

File: packages/@okta/vuepress-site/docs/reference/api/apps/index.md [hunk 3]
```diff
@@ -1328,7 +1328,7 @@ Adds an OAuth 2.0 client application. This application is only available to the
...
+* `issuer_mode` <ApiLifecycle access="ea" /> is visible if the Custom URL Domain feature is enabled. If the feature is enabled, you can set a custom domain URL in the settings for an OpenID Connect token in an app, and this property is returned in the appropriate responses. After the feature is enabled, the default value for new apps is `CUSTOM_URL`. For existing apps, the default remains `ORG_URL`. You can also update this to `DYNAMIC` if the Custom URL Domain feature is enabled. If it is set to `DYNAMIC` the request for the ID token made from Custom Domain will have the Custom URL as the issuer and the request for the ID token made from Okta Org Domain will have the Org URL as the issuer. After the feature is enabled, the default value for new apps is `DYNAMIC`. You can change the value using the API or administrator user interface. To enable the Custom URL Domain feature and `DYNAMIC` `issuerMode` feature, contact [Support](https://support.okta.com/help/open_case).
```
Comment (yuliu-okta) [inline packages/@okta/vuepress-site/docs/reference/api/apps/index.md:1331]: The original EA flag is for Custom URL Domain, I guess it is already GA, so we can remove that tag. But `DYNAMIC` is an ENG flag, do we doc ENG flag in API doc @susanharper-okta ?
Also `DYNAMIC` mode does not need Custom URL Domain feature.

**Review by yuliu-okta — APPROVED** [review]

**Pull request merged**